Add blog post about Kafka Exporter (#103)
strimzi/strimzi.github.io: _posts/2019-10-14-improving-prometheus-metrics.md

Strimzi just integrates it into our own Kafka deployments.
Kafka Exporter connects to Kafka as a client and collects different information about topics, partitions and consumer groups.
It then exposes this information as a Prometheus metric endpoint.

Deploying Kafka Exporter with Strimzi is easy.
All you need to do is to add it into a `Kafka` custom resource:

```yaml
apiVersion: kafka.strimzi.io/v1beta1
kind: Kafka
metadata:
  name: my-cluster
spec:
  # ...
  kafkaExporter: {}
```

Strimzi will create a new deployment with the Kafka Exporter and configure it to connect to the Kafka cluster.
You do not need to create any certificates or configure where it should connect.
But if you need, you can configure the consumer groups or the topics which it should monitor:

```yaml
apiVersion: kafka.strimzi.io/v1beta1
kind: Kafka
metadata:
  name: my-cluster
spec:
  # ...
  kafkaExporter:
    topicRegex: "important-topic-.*"
    groupRegex: "important-group-.*"
```

You can find all the available configuration option in our [API reference](https://strimzi.io/docs/latest/full.html#type-KafkaExporterSpec-reference).

> Kafka Exporter doesn't provide only the consumer lag metric.

Reviewer: ```suggestion
Kafka Exporter doesn't only provide the consumer lag metric.
```

It also provides a lot of additional details about consumer groups and topics.
* Information about message consumption rates.
* Latest offsets per consumer group.
* Latest and oldest offset for topics.
* Information about partitions which do not have their leader on the preferred node.
All these metrics can be used in alerts and dashboards.

[![Kafka Exporter dashboard]({{ "/assets/2019-10-14-kafka-exporter-dashboard.png" }})](/assets/2019-10-14-kafka-exporter-dashboard.png)

Once you have the Kafka Exporter deployed, you can start scraping the metrics it provides.
We also provide a new Grafana dashboard and alert rules which work with the new metrics.
In 0.14.0 the dashboard we have is fairly basic.

> In 0.15.0 we will release an improved Grafana dashboard which uses more of the different metrics provided by the Kafka Exporter.

Reviewer: ```suggestion
For 0.15.0 we are working on an improved Grafana dashboard which uses more of the different metrics provided by the Kafka Exporter.
```

Author: This is already done. I think I would prefer the keep my form to me this more clear so that the users can go and use the master version.

You can find the [sample Prometheus Alerts](https://github.com/strimzi/strimzi-kafka-operator/blob/master/metrics/examples/prometheus/install/prometheus-rules.yaml) and the [Grafana dashboard](https://github.com/strimzi/strimzi-kafka-operator/blob/master/metrics/examples/grafana/strimzi-kafka-exporter.json) on our GitHub.

## Conclusion

It is important for us to provide production ready Kafka.

> And monitoring is a big part of it.

Reviewer: ```suggestion
And effective monitoring is a big part of that.
```

> Kafka Exporter is an important improvement to our monitoring capabilities of Strimzi.

Reviewer: ```suggestion
Kafka Exporter is an important improvement to the monitoring capabilities of Strimzi.
```

Alertmanager will be then responsible for managing these alerts.
It will handle duplicates, silencing, inhibition and aggregation of alerts and send out notification to your system of choice.
You can have your alerts sent to many different notification channels such as emails, Slack, PagerDuty and more.

## Apache Kafka and Prometheus

Apache Kafka doesn't support Prometheus metrics natively by default.
Like many JVM based applications it uses [JMX (Java Management Extensions)](https://en.wikipedia.org/wiki/Java_Management_Extensions) for exposing metrics.
But since our aim in the Strimzi project is to offer a Kubernetes-native experience when running Apache Kafka, we want to have the metrics exposed as a Prometheus endpoint.
To help with that, we use the [Prometheus JMX Exporter](https://github.com/prometheus/jmx_exporter) project which takes the JMX metrics and exposes them as Prometheus endpoint.
Apart from integrating the JMX Exporter we also provide Grafana dashboards and sample Prometheus alerting rules which you can use and adapt to your own needs.

[![Kafka dashboard]({{ "/assets/2019-10-14-kafka-dashboard.png" }})](/assets/2019-10-14-kafka-dashboard.png)

> To enable the JMX Exporter in Strimzi, you have to specify the `metrics` section in the Strimzi custom resources.

Reviewer 2: maybe we can explicitly naming `Kafka` and `KafkaConnect`?
